File: Python/8kyu/8kyu-grasshopper-summation.py
# Grasshopper - Summation
# Language: Python
# Difficulty: 8 kyu

# Solution 1
def summation(num):
    return sum(list(range(1, num + 1, 1)))

# Solution 2 (same, but more concise)
def summation(num):
    return sum(range(1, num + 1))
    
# Tests
import codewars_test as test
from solution import summation

@test.describe('Fixed tests')
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(summation(1), 1)
        test.assert_equals(summation(8), 36)
        test.assert_equals(summation(22), 253)
        test.assert_equals(summation(100), 5050)
        test.assert_equals(summation(213), 22791)

File: R/8kyu/8kyu-grasshopper-summation.r
# Grasshopper - Summation
# Language: R
# Difficulty: 8 kyu

# Solution 1
summation <- function(n) {
    sum(seq(1, n))
}

# Solution 2
summation <- function(n) {
  sum(1:n)
}


# Tests
test_that('basic tests', {
  expect_equal(summation(1), 1)
  expect_equal(summation(8), 36)
  expect_equal(summation(22), 253)
  expect_equal(summation(100), 5050)
  expect_equal(summation(213), 22791)
})